android/settings.gradle

```diff
@@ -1,3 +1,10 @@
 rootProject.name = 'RDSApp'
 apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
 include ':app'
+
+if (settings.hasProperty("newArchEnabled") && settings.newArchEnabled == "true") {
+    include(":ReactAndroid")
+    project(":ReactAndroid").projectDir = file('../node_modules/react-native/ReactAndroid')
+    include(":ReactAndroid:hermes-engine")
+    project(":ReactAndroid:hermes-engine").projectDir = file('../node_modules/react-native/ReactAndroid/hermes-engine')
```

Comment:
This line is setting the project directory for the ReactAndroid:hermes-engine project to the path ../node_modules/react-native/ReactAndroid/hermes-engine. However, this path is relative to the current directory, which is the ../node_modules/react-native/ReactAndroid directory. This means that the project directory for the ReactAndroid:hermes-engine project will be set to the current directory, which is not the correct location.

Reply:
What you're saying would be correct except for the fact that when this script is taken into account is while doing any operation via gradle like gradlew clean, gradlew assembleRelease, etc. 
When gradle operations run, the current directory is always /android. So, if current directory is android then to get to hermes engine it will need to do cd .. (goto parent directly which also happens to be the root directory) and then into node modules.
This change came into effect due the the changes suggested by React Native upgrade helper. 
Please let me know if you have any suggestions or queries.